src/config/scrivitoSites.ts

```diff
@@ -26,30 +34,29 @@ export function baseUrlForSite(siteId: string): string | undefined {
 export function siteForUrl(
   url: string,
 ): { baseUrl: string; siteId: string } | undefined {
-  const neoletterMailingsBaseUrl = baseUrlForSite(NEOLETTER_MAILINGS_SITE_ID)
-  if (neoletterMailingsBaseUrl && url.startsWith(neoletterMailingsBaseUrl)) {
-    return {
-      baseUrl: neoletterMailingsBaseUrl,
-      siteId: NEOLETTER_MAILINGS_SITE_ID,
-    }
-  }
-
-  const language = /\b\/([0-9a-f]{32}\/)?(?<lang>[a-z]{2})([?/]|$)/.exec(url)
-    ?.groups?.lang
-
-  const siteId = allWebsites()
-    .and('_language', 'equals', language || null)
-    .first()
-    ?.siteId()
-
-  const baseUrl = siteId && baseUrlForSite(siteId)
-  if (baseUrl) return { baseUrl, siteId }
+  return Obj.onAllSites()
+    .where('_path', 'equals', '/')
+    .toArray()
```

Comment:
This is a performance problem. 

Currently, the callback loads at most 1 Obj (the current site roots).
With this PR, the callback now loads _all_ site roots, and then filters in memory.

This callback is on a "hot path", i.e. 
* it blocks the application, i.e. with this change more data needs to loaded on each page, before it can become visible/interactive
* it needs to be executed for every URL. This means when we later add pre-rendering, each page's contentDump will includes all site roots (as opposed to just the page's own site root, as usual).

So I'd rather find some way to keep it loading just a single Obj, instead of all site roots.